### DISCUSSION TITLE
why the mesh data get changed

### DISCUSSION THREAD

**Discussion by Sherman-c** [Jyers/Marlin#1102] [General Discussion]
I got 1.3.4 installed on my Ender3 V2, yesterday I went through 5x5 levelling with BL touch, not HS, after that I sent M420 V1, it returned the table below

Recv:       0      1      2      3      4
Recv:  0 -0.004 +0.004 +0.019 +0.078 +0.094
Recv:  1 -0.012 -0.003 +0.006 +0.064 +0.087
Recv:  2 -0.019 -0.006 -0.007 +0.039 +0.061
Recv:  3 -0.015 -0.002 +0.005 +0.049 +0.077
Recv:  4 -0.009 +0.016 +0.031 +0.082 +0.089

After 1 print, I power off the printer, didn't touch / change anything, today when I powered it backed on, again made no change whatsoever, I sent M420 V, I got this 
Recv:       0      1      2      3      4
Recv:  0 -0.002 -0.003 +0.013 +0.070 +0.091
Recv:  1 -0.013 -0.004 +0.000 +0.057 +0.088
Recv:  2 -0.026 -0.011 -0.007 +0.041 +0.057
Recv:  3 -0.013 -0.003 +0.003 +0.048 +0.072
Recv:  4 -0.007 +0.016 +0.022 +0.077 +0.089

although the different was small, only +/- 0.003 on average.  I just wonder why those values get changed.  The motherboard could not handle 2 decimal points properly?  32bits board should handle 2 decimal points easily.

PS i did restored to default before I started levelling.

Regards
Sherman

**Comment by gothcha** [comment]
The variation is likely caused by the storage of the values in the EEPROM. To save space only 16bits are used to represent the number. The value is scaled (typ x1000 but this can be varied) and saved as an integer, then divided on load. As a result the number of accurate decimal places is reduced as you have found.

**Reply by Sherman-c** [reply]
Thanks for explanation, that could explained why I see the value different.  On top of that, I done levelling again, I swear I did not touch the bed, this time  I got +/- 0.05 different.  I guess I have some hardware issue to deal with.  

Cheers
Sherman

**Reply by gothcha** [reply]
Try the repeatability test to check the BLTouch. 0.05mm is a very tiny variation so I wouldn’t be overly concerned.
